### ISSUE TITLE
How to become Repository

### ISSUE THREAD

**Issue by qjhart** [ucd-library/ror-extend-demo#5]
If we decide to include a ROR data repository for our data, we need to do a number of things:

1) Rename the repo

Proposals: *ror-extensions*, *ror-plus* , I don't have a good idea  

2) Define a structure for inputs.  https://github.com/perma-id/w3id.org is a good example of a project that uses github as a mechanism for storing data
3) Define the format for inputs 

Maybe just require an index.md file and then way we will only read *.jsonld files? There is still the matter of deciding whether the organizations can assign their own graph ids, or if we rewrite  these? 
I can see  points for both ways.

4) Reorganize our site so that it's more data centric.

**Comment by arthurpsmith** [comment]
Rename or create a new one? Maybe we should use the ror id as the name for the json-ld files? i.e. I see you did that for ucd - '05rrcem69.jsonld' ?

**Comment by qjhart** [comment]
@Vensberg / @arthurpsmith , I have added a new branch `re-org` with some proposed modifications. for the structure.  Basically, it moves the jsonld from examples to orgs, and then has slots for contributions there.    Please let me know what you think.

The structure is below, in addition we have `context` for jsonld context files, I'm working on the ror.owl for inference rules (e.g. child is inverse of parent).  the `vocabularies` should have our controlled vocabularies. 

I'm still debating Arthur's idea about usein the rorid for the names.  I'm leaning to allowing different orgs to do that if they want, but also not require it.  The idea being that we expect that in the beginning other organizations may fill in proxy structures, and we want to allow that, so there may not be an official tree, but there is some tree.  Also, you'll see that Carolyn's entries actually reference a lot of different RORs in her contributions.  

So I used dns names as a default, but even that's confusing.  Again, Carolyn has contributions (in cfa.harvard.edu) but they aren't actually related to harvard.  Maybe requiring some standard index.md file is the way to go here.


├── context
│   ├── grid.jsonld
│   └── vivo.jsonld
├── index.md
├── orgs
│   ├── cfa.harvard.edu
│   │   ├── cnr_it.jsonld
│   │   ├── dod_us.jsonld
│   │   ├── doe_us.jsonld
│   │   ├── index.md
│   │   ├── maxplank.jsonld
│   │   └── ras_ru.jsonld
│   ├── harvard.edu
│   │   ├── harvard.jsonld
│   │   ├── index.md
│   ├── ror.org
│   │   ├── README.org
│   │   └── ror.zip
│   └── ucdavis.edu
│       ├── 05rrcem69.jsonld
│       ├── README.md
├── rore
├── ror.owl
└── vocabularies
    ├── CIPCode2020.ttl
    └── README.md

**Comment by arthurpsmith** [comment]
In general the context, vocabularies, ror.owl stuff sounds fine to me, though there may be some specifics we should discuss further.
On the organization under "orgs" - I think you're mixing up provenance of an organization's data with the organization itself; ideally we would hope the organization provides its own data, but in practice for the vast majority of cases to start with that's unlikely (see VIVO); additionally we don't have a mechanism to verify accountability on that - the best we can hope for using this github repository mechanism is to identify the github users who contribute data.
Secondly, I really don't like using DNS name here. What ties the DNS name to the ROR id? Does every organization in ROR have a DNS name? What if we want to add entries for historical organizations? What about organizations that change their DNS (often happens when the name changes)?
I think a flat structure is simplest - orgs/&lt;rorid&gt;.jsonld - or if we really need a tree because we can't handle 100,000 files in one directory, then orgs/&lt;rorid 1st 2 chars&gt;/&lt;rorid&gt;.jsonld or something like that? That way the location of the extension file is completely derivable from the ror id itself, nothing further is needed.

**Comment by arthurpsmith** [comment]
Although, it occurs to me maybe the repository layout doesn't matter so much if what we are really providing is a service on top of it - that service could index all the files by ROR id to make them findable, so you'd just go to ror-extension.org/&lt;ror-id&gt; or something like that and get what you want, without worrying about exactly where the file came from?

**Comment by qjhart** [comment]
Yeah, so the reason that I separated out the orgs into directories was specifically for help in managing the commits into those locations.  Somehow to help manage pull-requests ,and being able to say, `hey this pull request is all over the place?  should they be making all these changes'  The general idea is some group stakes a claim on a directory in the orgs/ part, and then that's where the pull requests are limited to. 

I guess you idea is more along the lines of everyone edits the same ROR extenstions, and we just use git blame or something to track those changes I guess that'd be okay as well.  but not what I was thinking.

The main difference is the former idea can allow for two authorities to present a structure for an organization, and they can both be there, while in the latter, they would just have to fight over the changes to a single file.

**Comment by qjhart** [comment]
Regarding your second notion, yes,  in the end I believe we;d have  a service on top of this that allowed a query interface.  Hopefully the service would be as simple as *import all these files into a linked-data database* so that anyone could easily build their own interface as well.
